lib/spack/spack/repo.py

```diff
@@ -1235,6 +1236,18 @@ def swap(repo_path):
     path = saved
 
 
+@contextlib.contextmanager
+def additional_repository(repository):
+    """Adds temporarily a repository to the default one.
+
+    Args:
+        repository: repository to be added
+    """
+    path.put_first(repository)
```

Comment:
No as we never assign `path` in this code block. The reference documentation is [here](https://docs.python.org/3/reference/simple_stmts.html#the-global-statement) if you are interested. The relevant part of it is:

> It would be impossible to assign to a global variable without global, although free variables may refer to globals without being declared global.

A "free-variable" is a variable that is being used, but not declared global or bound to the current code block. The [reference](https://docs.python.org/3/reference/executionmodel.html) is in the "Execution model" part of python docs:

> If a variable is used in a code block but not defined there, it is a free variable.